packages/server-wallet/docs/architecture.md

```diff
@@ -34,39 +34,56 @@ A state channel wallet has the following _runtime dependencies_:
-                                                      +------------------+
+       +-------------------+                      |     +-------------+
+       |                   |                      |     |             |
+       |                   |                      |     |             |
+       |   CHAIN-SERVICE   <----------------------------> BLOCKCHAIN  |
+       |                   |                      |     |             |
+       |                   |                      |     |             |
+       +---------^---------+                      |     +-------------+
+                 |                                |                    
+                 |                                |                    
+                 |                                |                    
+                 |                                |                    
+                 |                                |                    
+          +------v-----+      +-------------+     |     +-------------+
+          |            |      |             |     |     |             |
+          |            |      |  MESSAGING  |     |     |    PEER     |
+          |            <------>             <----------->             |
+          |            |      |   SERVICE   |     |     |  MESSAGING  |
+          |            |      |             |     |     |             |
+          |            |      +-------------+     |     |   SERVICE   |
+          |            |                          |     |             |
+          |            |                          |     +-------------+
+          |   WALLET   |                          |                    
+          |            |                          |     +-------------+
+          |            |         API Call         |     |             |
+          |            <---------------------------------             |
+          |            |                          |     | APPLICATION |
+          |            |         API Event        |     |             |
+          |            |-------------------------------->             |
+          |            |                          |     |             |
+          |            |                          |     +-------------+
+          +------^-----+                                               
+                 |                                                     
+                 |                                                     
+                 |                                                     
+                 |                                                     
+                 |                                                     
+       +---------v--------+                                            
+       |      ENGINE      |                                            
+       |                  |                                            
+       |   +-----------+  |                                            
```

Comment:
Currently both the `wallet` and `engine` access the `store`

Reply:
What does the `wallet` use the store for?

Reply:
The most common use case is updating chain data in the `store` from `ChainEvent`s from the chain service before cranking the engine.

Reply:
is that the _only_ use case?

In my head, I expected that the wallet would just push a chain event into the engine, just like it pushes a message into the engine. In that world, a wallet would not need a store.

Reply:
da76ba062b78d4f8c91fcd897507f0a1a43b19c0